### PR TITLE
a8-review

### PR DESCRIPTION
**Pull request by swatikadivar**

### CODE REVIEW

File: aesd-char-driver/aesd-circular-buffer.c [hunk 1]
```diff
@@ -64,25 +65,31 @@ struct aesd_buffer_entry *aesd_circular_buffer_find_entry_offset_for_fpos(struct
 * Any necessary locking must be handled by the caller
 * Any memory referenced in @param add_entry must be allocated by and/or must have a lifetime managed by the caller.
 */
-void aesd_circular_buffer_add_entry(struct aesd_circular_buffer *buffer, const struct aesd_buffer_entry *add_entry)
+char* aesd_circular_buffer_add_entry(struct aesd_circular_buffer *buffer, const struct aesd_buffer_entry *add_entry)
```
Comment (swatikadivar) [inline aesd-char-driver/aesd-circular-buffer.c:68]: @lnxblog what would be better here instead of just char * ?

File: aesd-char-driver/main.c [hunk 2]
```diff
@@ -47,22 +53,83 @@ int aesd_release(struct inode *inode, struct file *filp)
...
 {
+	struct aesd_buffer_entry *entry;
+	size_t byte_offset,tx_size;
 	ssize_t retval = 0;
+	const char *source;
 	PDEBUG("read %zu bytes with offset %lld",count,*f_pos);
 	/**
 	 * TODO: handle read
 	 */
+ 	if (mutex_lock_interruptible(&aesd_device.cb_mutex))	// obtain lock
```
Comment (swatikadivar) [inline aesd-char-driver/main.c:64]: @lnxblog what mutex_lock_interruptible return if the lock is acquired?

**Review by swatikadivar — COMMENTED** [review]
nice work!

File: aesd-char-driver/aesd-circular-buffer.c (hunk 1 above)
Comment (lnxblog) [reply aesd-char-driver/aesd-circular-buffer.c:68]: void*

**Pull request merged**